zulipterminal/model.py

```diff
@@ -559,6 +560,50 @@ def update_private_message(self, msg_id: int, content: str) -> bool:
         display_error_if_present(response, self.controller)
         return response["result"] == "success"
 
+    def toggle_topic_resolved_status(self, stream_id: int, topic_name: str) -> bool:
+        if self.initial_data.get("realm_allow_community_topic_editing", None):
+            self.controller.report_error(" Editing topic is disabled")
+        else:
+            request: Dict[str, Any] = {
+                "anchor": "newest",
+                "num_before": 1,
+                "num_after": 0,
+                "narrow": [
+                    {
+                        "operator": "stream",
+                        "operand": self.stream_dict[stream_id]["name"],
+                    },
+                    {"operator": "topic", "operand": topic_name},
+                ],
+            }
+            response = self.client.get_messages(request)
+            if response["messages"] != []:
+                time_since_msg_sent = time.time() - response["messages"][0]["timestamp"]
+                edit_time_limit = self.initial_data[
+                    "realm_community_topic_editing_limit_seconds"
```

Comment:
Didn't know topics had a time limit. Then why do we show `"Only topic editing allowed. Time limit for editing the message body has been exceeded."` when `realm_message_content_edit_limit_seconds` is crossed?

Reply:
Who can resolve such a topic then, only admins? I assume it would be a future feature for ZT then? Probably comment it?

Reply:
We haven't added support for 'community' topic editing in the regular edit feature, so we give a notice there which represents the feature we support.